BSP/sharpMemoryLCD/LS013B7DH03.c

```diff
@@ -88,18 +88,19 @@ static bool LCDupdateDisplay(uint8_t screenLine, uint8_t *pixelBuf, uint16_t nbB
 // when ST will finally change the HAL_SPI_Transmit API to const
 {
     bool returnSuccess = true;
+    uint8_t nbScreenLines = (nbBytes*NB_BIT_PER_BYTE)/SCREEN_WIDTH;
 
     // check the data fits into screen and ends on line boundary
-    if(screenLine>0 && (screenLine+(nbBytes*8)/SCREEN_WIDTH)<=SCREEN_HEIGHT && (nbBytes%(SCREEN_WIDTH/NB_BIT_PER_BYTE))==0)
+    if(screenLine>0 && (screenLine+nbScreenLines-1)<=SCREEN_HEIGHT && (nbBytes%(SCREEN_WIDTH/NB_BIT_PER_BYTE))==0)
```

Comment:
No change requested.
In this case I like to introduce bool vars. This way no comments are needed, and the intent is clear 
```
bool lineFitVertically = screenLine>0 && (screenLine+nbScreenLines-1)<=SCREEN_HEIGHT;
bool lineFitHorizontally = (nbBytes%(SCREEN_WIDTH/NB_BIT_PER_BYTE))==0;
```